substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/snippets/ReflectionPlugins.java

```diff
@@ -267,6 +278,58 @@ private static boolean processGetField(GraphBuilderContext b, ResolvedJavaMethod
         return false;
     }
 
+    private static boolean processGetFields(GraphBuilderContext b, ResolvedJavaMethod targetMethod, Receiver receiver,
+                                           SnippetReflectionProvider snippetReflection, boolean declared, boolean analysis, boolean hosted) {
+        if (receiver.isConstant()) {
+            Class<?> clazz = snippetReflection.asObject(Class.class, receiver.get().asJavaConstant());
+
+            String target = clazz.getTypeName() + ".getDeclaredFields()";
+            try {
+                Field[] fields = declared ? clazz.getDeclaredFields() : clazz.getFields();
+                if (fields.length == 0) {
+                    return false;
+                }
```

Comment:
Please refactor the code below, until the `pushConstant()` call, into a generic `getIntrinsicArray()` method. It can be reused for intrinsifying calls to `getMethods()`/`getDeclaredMethods()`/`getConstructors()`/`getDeclaredConstructors()` as well. (Feel free to extend this PR with those implementations too.)